lib/Mailer/WordPress/WordPressMailer.php

```diff
@@ -51,6 +51,10 @@ public function send() {
     ];
 
     $sendWithMailer = function ($mailer) use ($email, $address, $extraParams) {
+      // we need to call Mailer::init() for every single WP e-mail to make sure reply-to is set
+      $replyTo = $this->getReplyToAddress();
+      $mailer->init(false, false, $replyTo);
```

Comment:
Since I couldn't find a way to just set the reply-to address as this is currently done inside \MailPoet\Mailer\Mailer::init(), I opted to call it here directly, passing the reply-to address (before init() would be called internally when the code calls \MailPoet\Mailer\Mailer::send() in the line below).

Besides being a weird design pattern to have to call init() multiple times just to set the reply-to address, the other problem that I see with this approach is that the same \MailPoet\Mailer\Mailer instance is shared across different parts of MailPoet. So setting the reply-to address here has the potential to impact other e-mails, and I don't think there is an easy way around this other than refactoring \MailPoet\Mailer\Mailer, which I'm hesitant to do as it is a core part of the plugin. For an example of how this change impacts other parts of the plugin, see the test https://github.com/mailpoet/mailpoet/pull/3632/files#diff-f92c8ae77731d86590a35535f19be126061e371e196cca07718583607a84b675R345 (make sure to see the code comment at the top of the test method).

I wonder if there is another way to fix this that I'm missing?

Reply:
> Besides being a weird design pattern to have to call init() multiple times just to set the reply-to address

Sorry, I am not sure I understand this note. Where do you call it multiple times? I can see it being called on this line, where is the other call?

> other problem that I see with this approach is that the same \MailPoet\Mailer\Mailer instance is shared across different parts of MailPoet

We can prevent that. All services are configured as shared and by default. But we can change it here https://github.com/mailpoet/mailpoet/blob/master/lib/DI/ContainerConfigurator.php#L248 and add `->setShared(false)` and a new instance will be created every time the class is used.

Reply:
> Sorry, I am not sure I understand this note. Where do you call it multiple times? I can see it being called on this line, where is the other call?

I guess the idea is that the `init`method is called per each callback execution.
In my opinion, it is not ideal, but it is better than refactoring the `\MailPoet\Mailer\Mailer` class.  So, I am okay with the current solution.

Reply:
Thanks for your comments, @pavel-mailpoet, and @lysyjan.

> Sorry, I am not sure I understand this note. Where do you call it multiple times? I can see it being called on this line, where is the other call?

I could have been more clear here. What I mean is that in a single request we sometimes send more than one WP e-mail and thus we would be calling init() multiple times. And that seems to me to be against what was intended when the Mailer class was designed, as inside send(), init() is only called if mailerInstance is not set (https://github.com/mailpoet/mailpoet/blob/03400bc4f7e55f1f41139d030d2cb6ddde7dc025/lib/Mailer/Mailer.php#L60).

> We can prevent that. All services are configured as shared and by default. But we can change it here https://github.com/mailpoet/mailpoet/blob/master/lib/DI/ContainerConfigurator.php#L248 and add ->setShared(false) and a new instance will be created every time the class is used.

That is good to know. I will explore this option. I imagined that the DI Container had a way to change this behavior, but I'm a bit reluctant to change it as there is a chance that this particular service is shared by design, and changing this behavior could break some other part of the code. Anyway, I will look into it as this should be enough to fix the last problem that exists in the implementation that I shared in this PR.

> I guess the idea is that the initmethod is called per each callback execution.
In my opinion, it is not ideal, but it is better than refactoring the \MailPoet\Mailer\Mailer class. So, I am okay with the current solution.

Thanks for your input. I will look into using setShared(false) as Pavel suggested as I believe this might be enough to make everything work. If there are no unwanted side effects, I also prefer this approach over refactoring the \MailPoet\Mailer\Mailer class.